### ISSUE TITLE
Batch Normalization in Neupy

### ISSUE THREAD

**Issue by EdwinGerman** [itdxer/neupy#201]
I want to know if exist any example using Batchnormalization in NeuPy. It could be interesting to know how to apply it in Training and Test(when the adjusted model is used to do a prediction) stage. 

Thank you very much.

**Comment by itdxer** [comment]
I don't think that such an example exists, but it would be useful to add one to the [BatchNorn](http://neupy.com/apidocs/neupy.layers.normalization.html#neupy.layers.normalization.BatchNorm) documentation. For training and test it can be used like this:

```python
network = algorithms.GradientDescent(...)  # layers with batch normalization
network.train(x_train, y_train, epochs=10)

with network.connection.disable_training_state():
    y_predicted = network.predict(x_test)
```

**Comment by EdwinGerman** [comment]
So in this case what is the correct code or both are the same:

1)  with Adamnet.connection.disable_training_state():
        y_predicted = Adamnet.predict(x_test_scaled)
or 

2)  y_predicted = Adamnet.predict(x_test_scaled)

Thank you very much for your response.

```
Adamnet = algorithms.Adamax(
        [layers.Input(x_train_scaled.shape[1]),

	 layers.BatchNorm(),
         layers.Tanh(10),
         layers.Dropout(0.01),
	 
	 #layers.BatchNorm(),
         #layers.Tanh(30),
         #layers.Dropout(0.5),

	 #layers.BatchNorm(),
         #layers.Tanh(30),
         #layers.Dropout(0.5),

	 layers.BatchNorm(),
         layers.Tanh(10),
         layers.Dropout(0.01),

	 layers.BatchNorm(),
         layers.Linear(1), ],

         shuffle_data = True,
         batch_size   = 100,
         verbose      = True,
         show_epoch   = 100,
         epsilon      = 1e-8,
    )
    if bool_k_fold:
        kf = KFold(n_splits=n_splits)
        kf.get_n_splits(x_train_scaled)
        for train_index, test_index in kf.split(x_train_scaled):
            x_train, x_val = x_train_scaled[train_index], x_train_scaled[test_index]
            y_train, y_val = y_train_scaled[train_index], y_train_scaled[test_index]
            Adamnet.train(x_train, y_train, x_val, y_val, epochs=num_epochs)
    else:
        x_train, x_val, y_train, y_val = train_test_split(
            x_train_scaled, y_train_scaled, train_size=0.90, shuffle=True,
        )
        Adamnet.train(x_train, y_train, x_val, y_val, epochs=num_epochs)

    plots.error_plot(Adamnet)

    **with Adamnet.connection.disable_training_state():
        y_predicted = Adamnet.predict(x_test_scaled)**

    y_predicted_denorm = y_scaler.inverse_transform(y_predicted)

    R1 = pearson_def(y_predicted_denorm, y_test_dataset)

    print(R1)
```

**Comment by itdxer** [comment]
Actually, in new versions `predict` method works independently from `disable_training_state`. The `predict` method is pre-compliled in the way that automatically assumes that prediction not in the training state. The `disable_training_state` methods useful only in lower-level APIs. You can find more information here: http://neupy.com/docs/layers/basics.html#compile-networks

**Comment by itdxer** [comment]
You can just ignore part with `with Adamnet.connection.disable_training_state():`, it will work in the same way without it. Sorry for the previous confusion

**Comment by EdwinGerman** [comment]
Thank you very much for the clarification. I really enjoy using this library.
